addons/website/models/mixins.py

```diff
@@ -197,7 +197,9 @@ def _default_is_published(self):
 
     def website_publish_button(self):
         self.ensure_one()
-        return self.write({'website_published': not self.website_published})
+        value = not self.website_published
+        self.write({'website_published': value})
+        return value
```

Comment:
For tracking purpose: since this method is called by this kind of button
![image](https://github.com/odoo/odoo/assets/30048408/9457364d-cf4b-4c83-8307-abf6c2ee8dac)

And since this method was changed to now return the state of the published value and not `False` all the time (return value of `write()` somehow), I was expecting a behavior change but there is none :+1: The framework is reloading the full record value after clicking on that so it reads the up to date published state and dislpay correctly the button

Reply:
\* was `True` all the time not `False` :wink: 

> I was expecting a behavior change but there is none 👍

Yep, there are multiple cases. That button but also inner buttons in one or two form views (not using the same widget), all working. The return value is just unused in those cases.

Reply:
Python wise `write()` return always `True` indeed, but I was referring to the console rpc returned value which is always false somehow:
![image](https://github.com/odoo/odoo/assets/30048408/5b57abf6-4398-4039-b68b-b289ab1e8f22)

I thought following your change it would return True/False depending of the published state, did not properly check as it's still returning always False, as it's hardcoded for `call_button`, I did not know that, I just checked.

![image](https://github.com/odoo/odoo/assets/30048408/9177c48c-aa82-4df2-b922-4f37faae4302)